src/game/savegame.c

```diff
@@ -251,84 +251,86 @@ void Savegame_ApplyLogicToStartInfo(int level_num)
     }
 }
 
-void Savegame_PersistGameToStartInfo(int level_num)
+void Savegame_ResetCurrentInfo(int level_num)
+{
+    RESUME_INFO *current = &g_GameInfo.current[level_num];
+    memset(current, 0, sizeof(RESUME_INFO));
+}
+
+void Savegame_CarryCurrentInfoToStartInfo(int32_t src_level, int32_t dst_level)
+{
+    g_GameInfo.start[dst_level] = g_GameInfo.current[src_level];
```

Comment:
I'm not sure if it's OK to assign to entire structs like this, or we need to memcpy